packages/util/src/parserFixPoint.ts

```diff
@@ -25,21 +25,24 @@ export interface ParserFixPointFactory {
 /// raw: substrate_fixed::types::I<upper>F<lower> as I<upper+lower>
 /// precision: 0..lower number bits in fractional part to process
 export const parserFixPoint: ParserFixPointFactory = function (upper, lower) {
-  const len = assertLength(upper, lower);
-  return (raw: BN, precision: number = lower): number => {
-    assert(raw.bitLength() <= len, 'Bit length is not equal to ' + len);
-    const bits: string = raw.toString(2, len);
-    const lowerBits: string = (lower > bits.length
-      ? bits.padStart(lower, '0')
-      : bits).slice(-lower, -1 * (lower - precision) || undefined);
-    const floatPart: number = lowerBits
-      .split('')
-      .reduce((acc, bit, idx) => {
-        acc = acc + (bit === '1' ? 1 / 2 ** (idx + 1) : 0);
-        return acc;
-      }, 0);
-    const upperBits: string = bits.slice(0, -lower);
-    const decimalPart: number = upperBits ? parseInt(upperBits, 2) : 0;
-    return decimalPart + (raw.isNeg() ? -floatPart : floatPart);
-  };
-}
+    const len = assertLength(upper, lower);
+    return (raw: BN, precision: number = lower): number => {
+        assert(raw.bitLength() <= len, "Bit length is not equal to " + len);
+
+        raw = raw.fromTwos(len);
```

Comment:
I guess this is the crucial line? It is a bit hard to see because of the mismatch in the indentation depths. But I accept in this case because the original indentation depth, is not the standard, and has been introduced by the creator of this repo. I did not know that we still have some leftovers from this time.